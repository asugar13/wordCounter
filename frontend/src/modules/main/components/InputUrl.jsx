import React from 'react';
import { useFormik } from "formik";
import * as Yup from 'yup';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { Button, TextField } from '@material-ui/core';



const InputUrl = ({ setSelectedUrl, urlHistory
}) => {
  const urlSchema = Yup.object().shape({
    url: Yup.string()
      .matches(/^(http|https):\/\/./gi, 'Enter a valid URL in HTTP/S format')
  });

  const formik = useFormik({
    initialValues: {
      url: "",
    },
    validationSchema: urlSchema,
    onSubmit: (values) => {
      let request = values.url
      setSelectedUrl(request)
    },
  })

  return <>
    <div className="selector">
      <form onSubmit={formik.handleSubmit}>
        <Autocomplete
          placeholder="Enter a valid URL"
          options={urlHistory?.map((option) => option.label).reverse()}
          onChange={(e, value) => formik.setFieldValue("url", value)}
          renderInput={(params) => (
            <>
              <TextField {...params}
                name="url"
                onChange={formik.handleChange}
                value={formik.url}
                label="Enter a valid URL" margin="normal" variant="outlined" />
            </>
          )}
        />
        <Button className="search" type="submit" variant="contained">Search</Button>

        {formik.errors.url && formik.touched.url ? <div className="validation">{formik.errors.url}</div> : null}
      </form>
    </div>
  </>
}

export default InputUrl