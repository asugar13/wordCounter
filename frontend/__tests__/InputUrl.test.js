import React from 'react';
import InputUrl from '../src/modules/main/components/InputUrl';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("<Input /> with empty urlHistory as prop", () => {
  const mockCallBack = jest.fn();
  const container = shallow(<InputUrl urlHistory={[]} />);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('should have one submit field', () => {
    expect(container.find('[type="submit"]').length).toEqual(1);
  });

  // it('should contain a Search button that responds to a click event', () => {
  //   container.find('[className="search"]').simulate('click')
  //   expect(mockCallBack.mock.calls.length).toEqual(1);
  // });
})