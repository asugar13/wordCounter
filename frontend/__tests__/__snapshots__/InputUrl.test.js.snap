// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> with empty urlHistory as prop should match the snapshot 1`] = `"<div class=\\"selector\\"><form><div class=\\"MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon\\" role=\\"combobox\\" aria-expanded=\\"false\\" placeholder=\\"Enter a valid URL\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" id=\\"undefined-label\\">Enter a valid URL</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\"><input type=\\"text\\" aria-invalid=\\"false\\" autoComplete=\\"off\\" name=\\"url\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\" aria-autocomplete=\\"list\\" autoCapitalize=\\"none\\" spellcheck=\\"false\\"/><div class=\\"MuiAutocomplete-endAdornment\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Clear\\" title=\\"Clear\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator\\" tabindex=\\"-1\\" type=\\"button\\" aria-label=\\"Open\\" title=\\"Open\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></span></button></div><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Enter a valid URL</span></legend></fieldset></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained search\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Search</span></button></form></div>"`;
